### PR TITLE
Option decorator

### PR DESCRIPTION
**Pull request by cooligus**

### CODE REVIEW

File: pkg/shared/option/compare.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"fmt"
+	"strings"
+)
+
+func Compare(userInput []string, realOptions []Option) ([]Option, error) {
+	for _, userOption := range userInput {
+		splitPos := strings.Index(userOption, "=")
+		name := userOption[:splitPos]
+		value := userOption[splitPos+1:]
```
Comment (sewe2000) [inline pkg/shared/option/compare.go:12]: What about?
``` go
name, value, foundOption := strings.Cut(userOption, "=")
```

File: internal/lua_tests/options_test.lua [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    #![options({
+        ageO?: int = 1,
+        position = "leader",
+        nameO?: string,
+        lastnameO?,
+        dog = {
+            hairType?,
+            age: int = 5
+        },
+        gender:{male;female}=male,
```
[the remark this answers is not in the document]
Comment (cooligus) [reply internal/lua_tests/options_test.lua:11]: `gender?:{male;female}=male` is what you want

**Review by BaderBC — CHANGES_REQUESTED** [review]

File: pkg/shared/option/option.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+		return "bool"
+	case List:
+		return "list"
+	case Struct:
+		return "struct"
+	case Enum:
+		return "enum"
+	default:
+		return "unknown"
+	}
```
[the remark this answers is not in the document]
Comment (cooligus) [reply pkg/shared/option/option.go:99]: Can you give me an example (sample code)?

File: cmd/guiApi/info.go
[the remark this answers is not in the document]
Comment (cooligus) [reply cmd/guiApi/info.go:1]: There is eg. displaying arrays

File: pkg/shared/option/option.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+func GetType(rawValue any) Type {
```
[the remark this answers is not in the document]
Comment (cooligus) [reply pkg/shared/option/option.go:102]: Can you explain?

[the remark this answers is not in the document]
Comment (BaderBC) [reply pkg/shared/option/option.go:102]: https://go.dev/tour/methods/16

File: internal/lua_tests/options_test.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (cooligus) [reply internal/lua_tests/options_test.lua:11]: Yes but it's empty string

File: internal/checker/interpreter.go [hunk 5]
```diff
@@ -38,6 +47,72 @@ func ExecuteLuaMain(script string, importLoopData *shared.ImportLoopData, ruleCo
...
+// TODO: allow to access array element by int not string
+func getLList(ruleOption option.Option, L *lua.LState) lua.LValue {
+	list := newLuaNamespace()
+
+	for i, value := range ruleOption.DefaultValue.([]string) {
+		list.AddField(strconv.Itoa(i), getAnyLValue(value, option.String))
+	}
+
+	return list.createTable(L)
+}
```
[the remark this answers is not in the document]
Comment (cooligus) [reply internal/checker/interpreter.go:84]: How to convert `[]string` to `lua.LValue`?

**Comment by BaderBC** [comment]
Please remove dependa bot from gh actions

**Review by BaderBC — APPROVED** [review]
Currently I don't really care about this project, so I approved it hoping this pr won't break anything.

**Pull request merged**